src/physics/euler.cpp

```diff
@@ -391,14 +378,8 @@ inline real2 Euler<dim,nstate,real>
 :: compute_entropy (const real2 density, const real2 pressure) const
 {
     real2 entropy = pressure * pow(density, -gam);
-    if (entropy > 0){
-        entropy = log( entropy );
-    } else {
-        entropy = BIG_NUMBER;
-        this->pcout << "WARNING: Entropy is not defined because " << std::endl 
-                    << "    pressure * pow(density, -gam) < 0 ." << std::endl
-                    << "    Setting entropy = BIG_NUMBER." << std::endl;
-    }
+    check_positive_quantity<real2>(entropy);
+    entropy = log(entropy);
     return entropy;
```

Comment:
did you intend to change the behavior? because now it will return `log(BIG_NUMBER)` instead of `BIG_NUMBER`

Reply:
which will be 100

Reply:
Hmm, I didn't catch this. The `do_nothing` case will behave quite differently now...

Reply:
One way I could modify the code is by adding (or modifying) `check_real_quantity()`. Then: 
* `entropy = log(entropy)`.  log(0) returns `-inf`, log(negative) returns `-nan`.
* `entropy = check_real_quantity(entropy)`, which would detect `nan` or `inf` and then pass to `this->handle_nonphysical_result()`.

Reply:
An if statement `if(entropy != BIG_NUMBER) { entropy = log(entropy);}` after calling `check_positive_quantity<real2>(entropy, "entropy");` could probably work.

Reply:
An if statement `if(entropy != BIG_NUMBER) { entropy = log(entropy);}` after calling `check_positive_quantity<real2>(entropy, "entropy");` could probably work.

Reply:
I have modified `check_positive_quantity` such that it returns a bool, which is used to return BIG_NUMBER if needed. Thanks @PranshulThakur for the suggestion!

I think this makes the behaviour the same as before. However it does lead to if-statements for nan detection in euler.cpp, which this PR sought to avoid. Let me know if there are any further suggestions for improvement.